File: docs_make/src/helpers/prep_input.R
library(tidyverse)

# Define the path to the input TSV file containing FASTA paths
input_file_path <- "~/SCIENCE/PROJECTS/+test/DoViP/inrefs.tsv"  # This file should have only one column, named 'inref'

# Read the TSV file into a tibble
input_df <- read.csv(file = input_file_path, header = TRUE, sep = "\t", stringsAsFactors = FALSE)

# Project-related parameters
input_df <- input_df %>%
  mutate(
    projtype = "singleworkflow",
    sample_set = "TEST_set",
    use_slurm = "false",
    continue = "false",
    stop_after_initial_predictors = "false",
    min_contig_length = "3000"
  )

# Genomad-related parameters
input_df <- input_df %>%
  mutate(
    genomad_signal = "do",
    genomad_res = "/path/to/previous/results",
    genomad_env = "conda_genomad",
    genomadDB_p = "path/to/genomad_db/genomad_db/",
    genomad_min_score = "0.7",
    genomad_sbatch_time = "2-0",
    genomad_cpus_per_task = "25",
    genomad_sbatch_mem = "20G"
  )

# DeepVirFinder-related parameters
input_df <- input_df %>%
  mutate(
    DVF_signal = "do",
    DVF_env = "conda_DVF",
    DVF_maxContigLen = "2099000",
    DVF_scoreTh = "0.7",
    DVF_pThreshold = "0.01",
    DVF_p = "path/to/DeepVirFinder/dvf.py",
    DVF_sbatch_time = "2-0",
    DVF_cpus_per_task = "15",
    DVF_sbatch_mem = "20G"
  )

# VirSorter2-related parameters
input_df <- input_df %>%
  mutate(
    virSorter2_signal = "do",
    virSorter2_res = "/path/to/previous/results",
    virSorter2_env = "conda_virsorter2",
    virSorter2DB_p = "path/to/virsorter-data/virsorter2/db/",
    virSorter2_high_confidence_only = "false",
    virSorter2_min_score = "0.5",
    virSorter2_sbatch_time = "4-0",
    virSorter2_cpus_per_task = "25",
    virSorter2_sbatch_mem = "20G"
  )

# VIBRANT-related parameters
input_df <- input_df %>%
  mutate(
    vibrant_signal = "do",
    vibrant_res = "/path/to/previous/results",
    vibrant_env = "conda_VIBRANT",
    vibrant_p = "path/to/VIBRANT/VIBRANT_run.py",
    vibrantDB_p = "/path/to/VIBRANT/databases/",
    vibrant_sbatch_time = "2-0",
    vibrant_cpus_per_task = "25",
    vibrant_sbatch_mem = "20G"
  )

# ViralVerify-related parameters
input_df <- input_df %>%
  mutate(
  viralVerify_signal ="do",
  viralVerify_res ="/path/to/previous/results",
  viralVerify_env ="conda_viralVerify",
  viralVerifyDB_p ="path/to/conda_viralVerify/DB/viralverifyDB_nbc_hmms.hmm",
  viralVerify_p ="path/to/conda/conda_viralVerify/viralVerify/bin/viralverify",
  viralVerfify_threshold ="7",
  viralVerify_sbatch_time ="2-0",
  viralVerify_cpus_per_task ="25",
  viralVerify_sbatch_mem ="20G"
  )
  
# CheckV-related parameters
input_df <- input_df %>%
  mutate(
  checkv_env ="conda_checkV",
  checkvDB_p ="path/to/CheckV/checkv-db-v1.5/",
  checkv_sbatch_time ="2-0",
  checkv_cpus_per_task ="2",
  checkv_sbatch_mem ="20G"
  )

# PhaTYP-related parameters
input_df <- input_df %>%
  mutate(
  phaTYP_env ="conda_PhaBOX",
  phaTYP_p ="/path/to/PhaBOX/",
  phaTYPdb_p ="path/to/PhaBOX/database/",
  phaTYPparam_p ="path/to/PhaBOX/parameters",
  phaTYP_sbatch_time ="2-0",
  phaTYP_cpus_per_task ="2",
  phaTYP_sbatch_mem ="20G"
  )

# NON-INTEGRATED viruses - final selection parameters
input_df <- input_df %>%
  mutate(
  NONInt_th_num_predictors_CheckV_NA ="3",
  NONInt_th_num_predictors_CheckV_AAIHighConf ="1",
  NONInt_th_completeness_CheckV_AAIHighConf ="30",
  NONInt_th_num_predictors_CheckV_AAIMediumConf ="2",
  NONInt_th_completeness_CheckV_AAIMediumConf ="10",
  NONInt_th_num_predictors_CheckV_HMM ="2",
  NONInt_th_completeness_CheckV_HMM ="10"
  )

# INTEGRATED viruses - final selection parameters
input_df <- input_df %>%
  mutate(  
  Int_th_num_predictors_CheckV_NA ="3",
  Int_th_num_predictors_CheckV_AAIHighConf ="1",
  Int_th_completeness_CheckV_AAIHighConf ="30",
  Int_th_num_predictors_CheckV_AAIMediumConf ="2",
  Int_th_completeness_CheckV_AAIMediumConf ="10",
  Int_th_num_predictors_CheckV_HMM ="2",
  Int_th_completeness_CheckV_HMM ="10"
)

# Write the data frame to a TSV file
out_p <- "~/SCIENCE/PROJECTS/+test/DoViP/output_inrefs_params.tsv"
write_tsv(input_df, out_p)

File: docs/helpers/prep_input.jl
using CSV
using DataFrames

input_file_path = "/data3/CLM_projs/TEST_Workflows/inDoViP/inrefs.tsv" #"path/to/file_with_input_fasta_paths.tsv"        # this file should have only one column, named 'inref'
input_df = CSV.read(input_file_path, DataFrame; delim='\t', header=true)

# project related parameters
input_df[!, :projtype] = fill("singleworkflow", nrow(input_df))
input_df[!, :sample_set] = fill("TEST_set", nrow(input_df))
input_df[!, :use_slurm] = fill("false", nrow(input_df))
input_df[!, :continue] = fill("false", nrow(input_df))
input_df[!, :stop_after_initial_predictors] = fill("false", nrow(input_df))
input_df[!, :min_contig_length] = fill("3000", nrow(input_df))

# genomad related parameters
input_df[!, :genomad_signal] = fill("do", nrow(input_df))
input_df[!, :genomad_res] = fill("/path/to/previous/results", nrow(input_df)) 
input_df[!, :genomad_env] = fill("conda_genomad", nrow(input_df))
input_df[!, :genomadDB_p] = fill("path/to/genomad_db/genomad_db/", nrow(input_df))
input_df[!, :genomad_min_score] = fill("0.7", nrow(input_df))
input_df[!, :genomad_sbatch_time] = fill("2-0", nrow(input_df))
input_df[!, :genomad_cpus_per_task] = fill("25", nrow(input_df))
input_df[!, :genomad_sbatch_mem] = fill("20G", nrow(input_df))

# DVF related parameters
input_df[!, :DVF_signal] = fill("do", nrow(input_df))
input_df[!, :DVF_env] = fill("conda_DVF", nrow(input_df))
input_df[!, :DVF_maxContigLen] = fill("2099000", nrow(input_df))
input_df[!, :DVF_scoreTh] = fill("0.7", nrow(input_df))
input_df[!, :DVF_pThreshold] = fill("0.01", nrow(input_df))
input_df[!, :DVF_p] = fill("path/to/DeepVirFinder/dvf.py", nrow(input_df))
input_df[!, :DVF_sbatch_time] = fill("2-0", nrow(input_df))
input_df[!, :DVF_cpus_per_task] = fill("15", nrow(input_df))
input_df[!, :DVF_sbatch_mem] = fill("20G", nrow(input_df))

# virSorter2 related parameters
input_df[!, :virSorter2_signal] = fill("do", nrow(input_df))
input_df[!, :virSorter2_res] = fill("/path/to/previous/results", nrow(input_df)) 
input_df[!, :virSorter2_env] = fill("conda_virsorter2", nrow(input_df))
input_df[!, :virSorter2DB_p] = fill("path/to/virsorter-data/virsorter2/db/", nrow(input_df)) 
input_df[!, :virSorter2_high_confidence_only] = fill("false", nrow(input_df))
input_df[!, :virSorter2_min_score] = fill("0.5", nrow(input_df))
input_df[!, :virSorter2_sbatch_time] = fill("4-0", nrow(input_df))
input_df[!, :virSorter2_cpus_per_task] = fill("25", nrow(input_df))
input_df[!, :virSorter2_sbatch_mem] = fill("20G", nrow(input_df))

# VIBRANT related parameters
input_df[!, :vibrant_signal] = fill("do", nrow(input_df))
input_df[!, :vibrant_res] = fill("/path/to/previous/results", nrow(input_df)) 
input_df[!, :vibrant_env] = fill("conda_VIBRANT", nrow(input_df))
input_df[!, :vibrant_p] = fill("path/to/VIBRANT/VIBRANT_run.py", nrow(input_df))
input_df[!, :vibrantDB_p] = fill("/path/to/VIBRANT/databases/", nrow(input_df))
input_df[!, :vibrant_sbatch_time] = fill("2-0", nrow(input_df))
input_df[!, :vibrant_cpus_per_task] = fill("25", nrow(input_df))
input_df[!, :vibrant_sbatch_mem] = fill("20G", nrow(input_df))

# viralVerify related parameters
input_df[!, :viralVerify_signal] = fill("do", nrow(input_df))
input_df[!, :viralVerify_res] = fill("/path/to/previous/results", nrow(input_df)) 
input_df[!, :viralVerify_env] = fill("conda_viralVerify", nrow(input_df))
input_df[!, :viralVerifyDB_p] = fill("path/to/conda_viralVerify/DB/viralverifyDB_nbc_hmms.hmm", nrow(input_df))
input_df[!, :viralVerify_p] = fill("path/to/conda/conda_viralVerify/viralVerify/bin/viralverify", nrow(input_df))
input_df[!, :viralVerfify_threshold] = fill("7", nrow(input_df))
input_df[!, :viralVerify_sbatch_time] = fill("2-0", nrow(input_df))
input_df[!, :viralVerify_cpus_per_task] = fill("25", nrow(input_df))
input_df[!, :viralVerify_sbatch_mem] = fill("20G", nrow(input_df))

#checkV related parameters
input_df[!, :checkv_env] = fill("conda_checkV", nrow(input_df))
input_df[!, :checkvDB_p] = fill("path/to/CheckV/checkv-db-v1.5/", nrow(input_df))
input_df[!, :checkv_sbatch_time] = fill("2-0", nrow(input_df))
input_df[!, :checkv_cpus_per_task] = fill("2", nrow(input_df))
input_df[!, :checkv_sbatch_mem] = fill("20G", nrow(input_df))  

#PhaTYP related parameters
input_df[!, :phaTYP_env] = fill("conda_PhaBOX", nrow(input_df))
input_df[!, :phaTYP_p] = fill("/path/to/PhaBOX/", nrow(input_df))
input_df[!, :phaTYPdb_p] = fill("path/to/PhaBOX/database/", nrow(input_df))
input_df[!, :phaTYPparam_p] = fill("path/to/PhaBOX/parameters", nrow(input_df))
input_df[!, :phaTYP_sbatch_time] = fill("2-0", nrow(input_df))
input_df[!, :phaTYP_cpus_per_task] = fill("2", nrow(input_df))
input_df[!, :phaTYP_sbatch_mem] = fill("20G", nrow(input_df))  

# Final thresholding related parameters for NON-INTEGRATED
input_df[!, :NONInt_th_num_predictors_CheckV_NA] = fill("3", nrow(input_df))
input_df[!, :NONInt_th_num_predictors_CheckV_AAIHighConf] = fill("1", nrow(input_df))
input_df[!, :NONInt_th_completeness_CheckV_AAIHighConf] = fill("30", nrow(input_df))
input_df[!, :NONInt_th_num_predictors_CheckV_AAIMediumConf] = fill("2", nrow(input_df))
input_df[!, :NONInt_th_completeness_CheckV_AAIMediumConf] = fill("10", nrow(input_df))
input_df[!, :NONInt_th_num_predictors_CheckV_HMM] = fill("2", nrow(input_df))
input_df[!, :NONInt_th_completeness_CheckV_HMM] = fill("10", nrow(input_df))


# Final thresholding related parameters for INTEGRATED
input_df[!, :Int_th_num_predictors_CheckV_NA] = fill("3", nrow(input_df))
input_df[!, :Int_th_num_predictors_CheckV_AAIHighConf] = fill("1", nrow(input_df))
input_df[!, :Int_th_completeness_CheckV_AAIHighConf] = fill("30", nrow(input_df))
input_df[!, :Int_th_num_predictors_CheckV_AAIMediumConf] = fill("2", nrow(input_df))
input_df[!, :Int_th_completeness_CheckV_AAIMediumConf] = fill("10", nrow(input_df))
input_df[!, :Int_th_num_predictors_CheckV_HMM] = fill("2", nrow(input_df))
input_df[!, :Int_th_completeness_CheckV_HMM] = fill("10", nrow(input_df))


out_p = "/data3/CLM_projs/TEST_Workflows/inDoViP/inrefs_params_test.tsv"#"path/to/output_inrefs_params.tsv"
CSV.write(out_p, input_df, delim='\t', header=true)


